util/chunk/codec.go

```diff
@@ -189,3 +189,138 @@ func init() {
 		allNotNullBitmap[i] = 0xFF
 	}
 }
+
+// Decoder decodes the data returned from the coprocessor and stores the result in Chunk.
+// How Decoder works:
+// 1. Initialization phase: Decode a whole input byte slice to Decoder.intermChk using Codec.Decode. intermChk is
+//    introduced to simplify the implementation of decode phase. This phase uses pointer operations with less CPU andF
+//    memory cost.
+// 2. Decode phase:
+//    2.1 Set the number of rows that should be decoded to a multiple of 8 greater than
+//        chk.RequiredRows() - chk.NumRows(), this can reduce the cost when copying the srcCol.nullBitMap into destCol.nullBitMap.
+//    2.2 Append srcCol.offsets to destCol.offsets when the elements is of var-length type. And further adjust the
+//        offsets according to descCol.offsets[destCol.length]-srcCol.offsets[0].
+//    2.3 Append srcCol.nullBitMap to destCol.nullBitMap.
+// 3. Go to step 1 when the input byte slice is consumed.
+type Decoder struct {
+	intermChk    *Chunk
```

Comment:
We may split this chunk into multiple chunks and reference the underlying data by slice cutting to avoid data copying.